src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/ArtworkFilterRefetch.tsx

```diff
@@ -12,7 +12,9 @@ interface Props {
 }
 
 class ArtworkGridRefetchContainerWrapper extends React.Component<Props> {
-  private isLoading = false // Used to prevent multiple in-flight requests
+  state = {
```

Comment:
I don't think we can do this, basically since `setState` is async (and a `render/componentDidUpdate` call seems to be happening more times than it should), when you check `this.state.isLoading` below, you may not get what you expect...eg, multiple in flight requests. So I think we shd change it back to an instance var.

Reply:
We'll need a way to retrigger a render so that the `isLoading` is passed into the child. I'll add a guard in `componentDidUpdate` so that it won't update if `this.state.isLoading` is true.

Reply:
(i actually initially had it this exact way and was noticing that there were _still_ multiple req getting kicked off when you clicked on a filter, and switched it to an instance var)

Reply:
That won't work, basically `this.state.isLoading` is going to be false twice in a row, b/c of the async nature of `setState`. You have to use an instance var b/c of that, unless that was fixed elsewhere or since then?

Basically, open the inspector and click on a medium filter, that should kick off 1 request (if it's not in the Relay cache already), i think with this solution there'll be at least 2.

Reply:
I just pushed -- could you maybe pull down and give it a look? Looking at logs only one req is going out on click, which seems correct.

Reply:
@mzikherman - actually scratch that, it is indeed sending out two requests.